src/drawer/pages/MyDrawer/MyDrawer.tsx

```diff
@@ -14,11 +16,24 @@ const Dummy = Array.from({ length: 12 }).map((_, index) => index);
 export type TabType = 'STAR' | 'MYDRAWER';
 
 export const MyDrawer = () => {
-  const [currentTab, setCurrentTab] = useState<TabType>('STAR');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initialTab = searchParams.get('tab');
+
+  const [currentTab, setCurrentTab] = useState<TabType>(getTabFromURL() || 'STAR');
+
+  function isTabType(tab: string): tab is TabType {
+    return tab === 'STAR' || tab === 'MYDRAWER';
+  }
```

Comment:
isTabType 함수의 매개변수인 `tab`의 타입을 null도 허용하게 수정하면 `getTabFromURL()` 함수 없이 `useState()`를 리팩토링 할 수 있을 것 같아요!

```suggestion
  const [currentTab, setCurrentTab] = useState<TabType>(
    isTabType(initialTab) ? initialTab : 'STAR'
  );

  function isTabType(tab: string | null): tab is TabType {
    return tab === 'STAR' || tab === 'MYDRAWER';
  }
```

Reply:
당신천재야